components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -4121,4 +4129,59 @@ protected void checkAccessControlPermission(APIIdentifier identifier) throws API
         return updatedWSDLContent;
     }
 
+    /**
+     * This method is used to get keys of custom attributes, configured by user
+     *
+     * @param userId user name of logged in user
+     * @return Array of JSONObject, contains keys of attributes
+     * @throws APIManagementException
+     */
+    public JSONArray getAppAttributesFromConfig(String userId) throws APIManagementException {
+
+        String tenantDomain = MultitenantUtils.getTenantDomain(userId);
+        int tenantId = 0;
+        try {
+            tenantId = getTenantId(tenantDomain);
+        } catch (UserStoreException e) {
+            handleException("Error in getting tenantId of " + tenantDomain, e);
+        }
+        JSONArray applicationAttributes;
+        JSONObject applicationConfig = APIUtil.getAppAttributeKeysFromRegistry(tenantId);
+        if (applicationConfig != null) {
+            applicationAttributes = (JSONArray) applicationConfig.get("Attributes");
```

Comment:
Here you can use the constant, APIConstants.ApplicationAttributes.ATTRIBUTE.

Reply:
Variable, APIConstants.ApplicationAttributes.ATTRIBUTE = "Attribute".
I have added a new variable in APIConstants.java for "Attributes".